assets/src/components/propertiesPanel/crowdingDiagram.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react"
+import { crowdingIcon } from "../../helpers/icon"
+import { Vehicle } from "../../realtime"
+
+const extendedOccupancyStatus = (vehicle: Vehicle): string => {
+  // GTFS-RT has an EMPTY occupancy status but we don't use it; an empty
+  // vehicle will have a MANY_SEATS_AVAILABLE status. Additionally, it's
+  // possible that the route this vehicle is on generally has crowding data,
+  // but it's not working at the moment for this vehicle in particular.
+
+  if (vehicle.load === null) {
+    return "NO_DATA"
+  }
+
+  if (vehicle.load === 0) {
+    return "EMPTY"
+  }
+
+  // If vehicle.load isn't null, vehicle.occupancyStatus shouldn't be either.
+  return vehicle.occupancyStatus!
+}
+
+const GenericCrowdingDiagram = ({
+  vehicle,
+  statusDescription,
+}: {
+  vehicle: Vehicle
+  statusDescription: string
+}) => {
+  const classModifier = statusDescription.toLowerCase().replace(" ", "-")
+
+  return (
+    <div className="m-crowding-diagram">
+      <div className="m-crowding-diagram__properties">
+        <span className="m-properties-list__property-label">
+          Riders onboard
+        </span>
+        <br />
+        {vehicle.load} riders / {vehicle.capacity} maximum
+        <br />
+        <span
+          className={`m-crowding-diagram__status-description m-crowding-diagram__status-description--${classModifier}`}
+        >
+          {statusDescription}
+        </span>
+      </div>
+      <div className="m-crowding-diagram__crowding-icon-wrapper">
+        {crowdingIcon(
+          `m-crowding-diagram__crowding-icon m-crowding-diagram__crowding-icon--${classModifier}`
+        )}
+      </div>
+    </div>
+  )
+}
+
+const NoDataCrowdingDiagram = () => (
```

Comment:
This is pretty similar to the generic component. Would it be reasonable to have this drawn by the generic instead of being separate? The css class and text is already dynamic there. It'd just need to put a couple of pieces in a conditional.